### PR TITLE
Printer refactor

### PR DESCRIPTION
**Pull request by Alexander-Duke**

### CODE REVIEW

File: com/company/PrintAndCats/Printer.java [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
-        System.out.println(greeting + cat.age + " years old " + cat.name);
+        System.out.println(cat.age + " years old " + cat.name);
 
     }
 
+    public void printCatWithGreeting(Cat cat, String greeting){
+        System.out.println(greeting + " " + cat.name);
+    }
+
+    //todo: create method that takes two arguments Cat and greeting message, and prints greeting message with cat name
```
Comment (viacheslaev) [inline com/company/PrintAndCats/Printer.java]: remove todo

File: com/company/PrintAndCats/Main.java [hunk 2]
```diff
@@ -12,10 +12,10 @@ public static void main(String[] args) {
```
Comment (viacheslaev) [inline com/company/PrintAndCats/Main.java:8]: 1. Create two cats initialized by constructor with arguments (name, age)
2. Make name and age private fields
3. Make getter/setter for name and age
4. Override toString Method to print name, age

File: com/company/PrintAndCats/Main.java [hunk 3]
```diff
@@ -12,10 +12,10 @@ public static void main(String[] args) {
 
 
         Printer p = new Printer();
-        p.greeting = "Your cat is ";
 
         p.printCat(blackCat);
         p.printCat(redCat);
+        p.printCatWithGreeting(blackCat,"Hello");
 
 //изучить паттерн singletone
```
Comment (viacheslaev) [inline com/company/PrintAndCats/Main.java]: remove comments

File: com/company/PrintAndCats/Printer.java [hunk 4]
```diff
@@ -1,11 +1,16 @@
 package com.company.PrintAndCats;
 
 public class Printer {
-    String greeting;
 
     public void printCat(Cat cat) {
-        System.out.println(greeting + cat.age + " years old " + cat.name);
+        System.out.println(cat.age + " years old " + cat.name);
```
Comment (viacheslaev) [inline com/company/PrintAndCats/Printer.java]: excess empty line

File: com/company/PrintAndCats/Printer.java [hunk 5]
```diff
@@ -1,11 +1,16 @@
...
 
     }
 
+    public void printCatWithGreeting(Cat cat, String greeting){
+        System.out.println(greeting + " " + cat.name);
```
Comment (viacheslaev) [inline com/company/PrintAndCats/Printer.java]: Change println to printF method and use formatting to get rid of + operators

File: com/company/PrintAndCats/Printer.java (hunk 1 above)
Comment (viacheslaev) [inline com/company/PrintAndCats/Printer.java]: remove todo

Comment (Alexander-Duke) [reply com/company/PrintAndCats/Printer.java]: Ok

File: com/company/PrintAndCats/Main.java (hunk 2 above)
Comment (Alexander-Duke) [reply com/company/PrintAndCats/Main.java:8]: Ok

File: com/company/PrintAndCats/Printer.java (hunk 5 above)
Comment (Alexander-Duke) [reply com/company/PrintAndCats/Printer.java]: Ok

**Review by viacheslaev — APPROVED** [review]
Good job

**Pull request merged**